centralized-sampling-tests/sample-apps/golang-http-server/application.go

```diff
@@ -0,0 +1,133 @@
+
+	"go.opentelemetry.io/contrib/propagators/aws/xray"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
+	sample "go.opentelemetry.io/contrib/samplers/aws/xray"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/sdk/trace"
+	oteltrace "go.opentelemetry.io/otel/trace"
+
+	"google.golang.org/grpc"
+)
+
+func getSampledSpanCount(name string, totalSpans string, attributes []attribute.KeyValue) int {
+	tracer := otel.Tracer(name)
+
+	var sampleCount = 0
+	totalSamples, _ := strconv.Atoi(totalSpans)
+	ctx := context.Background()
+
+	for i := 0; i < totalSamples; i++ {
+		_, span := tracer.Start(ctx, name, oteltrace.WithSpanKind(oteltrace.SpanKindServer), oteltrace.WithAttributes(attributes...))
+
+		if span.SpanContext().IsSampled() {
+			sampleCount++
+		}
+
+		span.End()
+	}
+
+	return sampleCount
+}
+
+func webServer() {
+	http.Handle("/", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		_, _ = w.Write([]byte("healthcheck"))
+	}))
+
+	http.Handle("/getSampled", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		userAttribute := r.Header["User"][0]
+		required := r.Header["Required"][0]
+
+		var attributes = []attribute.KeyValue{
+			attribute.KeyValue{"http.method", attribute.StringValue(r.Method)},
+			attribute.KeyValue{"http.url", attribute.StringValue("http://localhost:8080/getSampled")},
+			attribute.KeyValue{"user", attribute.StringValue(userAttribute)},
+			attribute.KeyValue{"http.route", attribute.StringValue("/getSampled")},
+			attribute.KeyValue{"required", attribute.StringValue(required)},
+			attribute.KeyValue{"http.target", attribute.StringValue("/getSampled")},
+		}
+
+		var totalSampled = getSampledSpanCount(r.Header["Service_name"][0], r.Header["Totalspans"][0], attributes)
+		_, _ = w.Write([]byte(strconv.Itoa(totalSampled)))
+	}))
+
+	http.Handle("/importantEndpoint", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		userAttribute := r.Header["User"][0]
+		required := r.Header["Required"][0]
+
+		var attributes = []attribute.KeyValue{
+			attribute.KeyValue{"http.method", attribute.StringValue("GET")},
+			attribute.KeyValue{"http.url", attribute.StringValue("http://localhost:8080/importantEndpoint")},
+			attribute.KeyValue{"user", attribute.StringValue(userAttribute)},
+			attribute.KeyValue{"http.route", attribute.StringValue("/importantEndpoint")},
+			attribute.KeyValue{"required", attribute.StringValue(required)},
+			attribute.KeyValue{"http.target", attribute.StringValue("/importantEndpoint")},
+		}
+
+		var totalSampled = getSampledSpanCount(r.Header["Service_name"][0], r.Header["Totalspans"][0], attributes)
+		_, _ = w.Write([]byte(strconv.Itoa(totalSampled)))
+	}))
+
+	listenAddress := os.Getenv("LISTEN_ADDRESS")
+	if listenAddress == "" {
+		listenAddress = "0.0.0.0:8080"
+	}
+	log.Println("App is listening on %s !", listenAddress)
+	_ = http.ListenAndServe(listenAddress, nil)
+}
+
+func start_xray() {
```

Comment:
Should this function return an error here so that it can be propogated back to main if startup fails?